sample_annotator/clients/submission_portal_client.py

```diff
@@ -0,0 +1,185 @@
+        return return_lis
+        
+            #Processing the json response from the API
+    #Will return a pandas dataframe to be used to make the biosample for the database
+    def process_json_response(self,response):
+        #There could be more than one type of sampledata in the sampleData lists, so we'll make a dataframe for each one. This also adds some flexibility and robustness
+
+        string_params = ['samp_name', 'id', 'analysis_type', 'sample_link', 'ecosystem', 'ecosystem_category', 'ecosystem_type', 'ecosystem_subtype', 'specific_ecosystem', 'extreme_event', 'fao_class', 'fire', 'flooding', 'oxy_stat_samp', 'prev_land_use_meth', 'soil_horizon', 'soil_texture_meth', 'tillage', 'water_content', 'water_cont_soil_meth', 'micro_biomass_meth', 'tot_nitro_cont_meth', 'samp_collec_device', 'samp_collec_method', 'rel_to_oxygen', 'collection_time', 'collection_date_inc', 'collection_time_inc', 'start_date_inc', 'start_time_inc', 'filter_method', 'experimental_factor_other', 'non_microb_biomass', 'non_microb_biomass_method', 'microbial_biomass_c', 'micro_biomass_c_meth', 'microbial_biomass_n', 'micro_biomass_n_meth', 'org_nitro_method', 'other_treatment', 'isotope_exposure']
+        text_params = ['source_mat_id','env_package', 'agrochem_addition', 'al_sat_meth', 'crop_rotation', 'cur_vegetation', 'cur_vegetation_meth', 'heavy_metals', 'heavy_metals_meth', 'horizon_meth', 'link_class_info', 'link_climate_info', 'local_class', 'misc_param', 'local_class_meth', 'previous_land_use', 'soil_type', 'soil_type_meth', 'ph_meth', 'tot_org_c_meth', 'salinity_meth', 'store_cond', 'geo_loc_name', 'sieving', 'biotic_regm', 'air_temp_regm', 'climate_environment', 'gaseous_environment', 'humidity_regm', 'light_regm', 'watering_regm']
+        quantity_params = ['slope_aspect', 'al_sat', 'annual_precpt', 'annual_temp', 'season_precpt', 'season_temp', 'slope_gradient', 'soil_text_measure', 'temp', 'microbial_biomass', 'carb_nitro_ratio', 'org_matter', 'org_nitro', 'tot_carb', 'tot_nitro_content', 'tot_org_carb', 'tot_phosp', 'phosphate', 'salinity', 'samp_store_temp', 'size_frac_low', 'depth', 'size_frac_up', 'samp_size', 'alt']
+        controlled_term_params = ['env_broad_scale', 'env_local_scale', 'env_medium', 'experimental_factor', 'growth_facil', 'samp_mat_process', 'chem_administration']
+        float_params = ['elev', 'ph']
+        time_params = ['collection_date']
+        geoloc_params = ['lat_lon']
+
+
+        sampledata = response['metadata_submission']['sampleData']
+        db_lis = []
+        for super_keys in sampledata.keys():
+            current_lis = sampledata[super_keys]
+            db = nmdc.Database()
+            id_counter = 0
+            nmdc_ids = self.mint_nmdc_ids(len(current_lis))
+
+            for samp_dic in current_lis:
+                param_dic = {}
+                for key in samp_dic.keys():
+
+                    if(samp_dic[key] == 'nan' or samp_dic[key] == 'null' or samp_dic[key] == None):
```

Comment:
Any reason why you're using parentheses in your `if` conditionals? It's not required if you're not trying to enforce an order of operations. If you had a combination of `or`, `and` and `()` groupings it would be useful to wrap `if` conditionals in parentheses, but I don't think it's necessary here.

Reply:
Theres no reason other than habit. I also work with a few other languages that do require parentheses, so I got into the habit of always using them. I can remove them if needed.